analyses/sex-prediction-from-RNASeq/run-sex-prediction-from-RNASeq.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# This script runs the sex-prediction-from-RNASeq analysis
+# Author's Name Bill Amadio 2019
+
+set -e
+set -o pipefail
+
```

Comment:
The reason this fails in continuous integration (CI) currently is because the current directory when it is being run in CI is the root of the directory. You can add the following here:
```sh
# This script should always run as if it were being called from
# the directory it lives in.
script_directory="$(perl -e 'use File::Basename;
  use Cwd "abs_path";
  print dirname(abs_path(@ARGV[0]));' -- "$0")"
cd "$script_directory" || exit
```
And this should fix the directory issues with CI and make sure it runs the same way no matter where it's called from.

Reply:
Another thing you might want to try in this shell script is using variables for the arguments you are making to the various R scripts. For example, the data where the test and training data lives is constant across the 3 scripts. 
If you wanted to change the directory name for the processed data, you would have to change it in three places instead of one place if you used a variable. You could add the following to the top of the script, along with any other arguments it'd be good to set as variables (e.g., model directory, filename lead):

```sh
# use a variable for the directory that contains the training/testing data
PROCESSED=processed_data
```
and then to use it, the lines below become:

```sh
Rscript --vanilla 01-clean_split_data.R --expression ../../data/pbta-gene-expression-kallisto.stranded.rds --metadata ../../data/pbta-histologies.tsv --output_directory $PROCESSED --filename_lead kallisto_stranded

Rscript --vanilla 02-train_elasticnet.R --input_directory $PROCESSED --output_directory models --filename_lead kallisto_stranded

Rscript --vanilla 03-evaluate_model.R --test_set_input_directory $PROCESSED --model_input_directory models --output_directory results --filename_lead kallisto_stranded
```

Also worth a thought as you revisit this - if you use a bash script to run everything, do you need `03-evaluate_model.R` to 'construct' the input file names using `paste`?